app/screens/settings/notification_mention/mention_settings.tsx

```diff
@@ -0,0 +1,192 @@
+import {t} from '@i18n';
+import UserModel from '@typings/database/models/servers/user';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const UPDATE_MENTION_PREF = 'UPDATE_MENTION_PREF';
+const INITIAL_STATE = {
+    firstName: false,
+    usernameMention: false,
+    channel: false,
+};
+type Action = {
+    type: string;
+    data: Partial<typeof INITIAL_STATE>;
+}
+const reducer = (state: typeof INITIAL_STATE, action: Action) => {
+    switch (action.type) {
+        case UPDATE_MENTION_PREF:
+            return {
+                ...state,
+                ...action.data,
+            };
+
+        default:
+            return state;
+    }
+};
+
+const mentionHeaderText = {
+    id: t('notification_settings.mentions.wordsTrigger'),
+    defaultMessage: 'Words that trigger mentions',
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        separator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            flex: 1,
+            height: 1,
+            marginLeft: 15,
+        },
+        area: {
+            paddingHorizontal: 16,
+        },
+        upperCase: {
+            textTransform: 'uppercase',
+        },
+    };
+});
+
+type MentionSectionProps = {
+    currentUser?: UserModel;
+    mentionKeys: string;
+}
+const MentionSettings = ({currentUser, mentionKeys}: MentionSectionProps) => {
+    const [{firstName, usernameMention, channel}, dispatch] = useReducer(reducer, INITIAL_STATE);
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    let mentionKeysComponent;
+    if (mentionKeys) {
+        mentionKeysComponent = (<Text>{mentionKeys}</Text>);
+    } else {
+        mentionKeysComponent = (
+            <FormattedText
+                id='notification_settings.mentions.keywordsDescription'
+                defaultMessage='Other words that trigger a mention'
+            />
+        );
+    }
+
+    const toggleChannelMentions = () => {
+        dispatch({
+            type: UPDATE_MENTION_PREF,
+            data: {
+                channel: !channel,
+            },
+        });
+    };
+    const toggleUsernameMention = () => {
+        dispatch({
+            type: UPDATE_MENTION_PREF,
+            data: {
+                usernameMention: !usernameMention,
+            },
+        });
+    };
+    const toggleFirstNameMention = () => {
+        dispatch({
+            type: UPDATE_MENTION_PREF,
+            data: {
+                firstName: !firstName,
+            },
+        });
+    };
+    const goToNotificationSettingsMentionKeywords = () => {
+        return Alert.alert(
+            'The functionality you are trying to use has not yet been implemented.',
+        );
+    };
+
+    return (
+        <Block
+            headerText={mentionHeaderText}
+            headerStyles={styles.upperCase}
+            containerStyles={styles.area}
+        >
+            <>
```

Comment:
Why the fragment? Block does not allow many children?

Reply:
corrected